### PR TITLE
planner: do not build MPP plan for scan with virtual columns

### PR DESCRIPTION
**Pull request by eurekaka**
### What problem does this PR solve?

Issue Number: close https://github.com/pingcap/tidb/issues/23886

Problem Summary:

Planner reports resolve index failure for MPP plan on virtual columns.

### What is changed and how it works?

What's Changed:

Avoid generating MPP plan when the scan contains virtual columns.

How it Works:

If we enable MPP plan for such kind of cases, we need to add an extra projection node above to prune the newly appended columns, as what we do for `copTask`, otherwise we would get wrong query results. However, for a MPP task, we may attach plan nodes such as projection, hash aggregation and hash join upon the scan node, which would change the schema of the plan tree, so we have no idea of the target schema that the plan tree should be projected to then. Unlike `copTask` in which the `PhysicalTableReader` is the counterpart of `DataSource`, so we can add the projection upon the `PhysicalTableReader`, while in `mppTask`, the `PhysicalTableReader` is acting more like a data receiver, instead of the counterpart of the `DataSource`.

Meanwhile, I am not sure whether MPP execution engine can handle plan snippets which contain virtual columns, since I noticed that in `addPushedDownSelectionToMppTask`, no MPP task would be produced if the pushed down filter contains virtual columns.

### Related changes

- Need to cherry-pick to the release branch

### Check List 

Tests 

- Unit test

Side effects

N/A

### Release note 

- Do not build MPP plan for scan with virtual columns

### CODE REVIEW

**Review by hanfei1991 — APPROVED** [review]

File: planner/core/find_best_task.go [hunk 1]
```diff
@@ -1495,6 +1495,11 @@ func (ds *DataSource) convertToTableScan(prop *property.PhysicalProperty, candid
 			// If ts is a single partition, then this partition table is in static-only prune, then we should not choose mpp execution.
 			return &mppTask{}, nil
 		}
+		origColsLen := len(ts.Columns)
+		ts.Columns = ExpandVirtualColumn(ts.Columns, ts.schema, ts.Table.Columns)
```
Comment (wjhuang2016) [inline planner/core/find_best_task.go:1499]: If the virtual column is built on constant, then len(ts.Columns) == origColsLen.
Is it expected to build the MPP in this case?

Comment (eurekaka) [reply planner/core/find_best_task.go:1499]: Nice catch.

**Comment by ichn-hu** [comment]
Hi @eurekaka the e2e-test's failing is not your fault, I'll help you redo the check

**Pull request merged**